tools/pio/post_esp32.py

```diff
@@ -41,9 +41,10 @@ def esp32_create_combined_bin(source, target, env):
     flash_freq = env.BoardConfig().get("build.f_flash", '40m')
     flash_freq = flash_freq.replace('000000L', 'm')
     flash_mode = env.BoardConfig().get("build.flash_mode", "dio")
-    if flash_mode == "qio":
+    memory_type = env.BoardConfig().get("build.arduino.memory_type", "qio_qspi")
```

Comment:
What memory type should I use for this in the board defs?

Reply:
Not only one. If you want to provide precompiled S3 firmwares, you have to provide all four flash/psram memory combinations. Dont know if these variants work for S3 without PSRAM too (are there any?).